### ISSUE TITLE
cmd/cgo: fails to bootstrap on illumos-amd64 builder with "undefined: ast.IndexListExpr"

### ISSUE THREAD

**Issue by bcmills** [golang/go#48863]
`greplogs --dashboard -md -l -e 'undefined: ast.IndexListExpr'`

[2021-10-08T02:09:58-e74db46/illumos-amd64](https://build.golang.org/log/1cf6bc8fdb56079fd3c8d76ff0f0976b67af0d94)
[2021-10-08T01:18:01-8c4ea31/illumos-amd64](https://build.golang.org/log/d8bc81bff7d8b86e41cc3e5633a01ac97d8cc156)
[2021-10-08T00:24:52-78d749f/illumos-amd64](https://build.golang.org/log/92d14620f03e0a0d59becd4572acb2b35ba7b3cb)
[2021-10-08T00:24:51-5780be4/illumos-amd64](https://build.golang.org/log/45dda8b74ebdadfe58358210494ef7a451b81c03)
[2021-10-08T00:18:29-7cef831/illumos-amd64](https://build.golang.org/log/204eace753061d59fede33df51a8f0e49022c247)
[2021-10-07T22:19:58-1ceb723/illumos-amd64](https://build.golang.org/log/8ac3a34ac77b918b6fb23f44043ad986e85f914f)
[2021-10-07T22:19:51-6436f5c/illumos-amd64](https://build.golang.org/log/28cd2045513d4a9c2f4507063aedc5bf3e67588c)
[2021-10-07T21:46:33-b69f823/illumos-amd64](https://build.golang.org/log/9496445f65f6bcc349376fe98663ca2cf4cdafe2)
[2021-10-07T21:39:17-6b4cf2b/illumos-amd64](https://build.golang.org/log/981c4e88817c22036e132278a95cce67616145ce)
[2021-10-07T20:37:43-ef2ebbe/illumos-amd64](https://build.golang.org/log/0bd78accea17cfe0b7554bbab732f9caaf4d4ddf)
[2021-10-07T20:02:29-0f52292/illumos-amd64](https://build.golang.org/log/cf05d699c5b39e01cf719ff3f5e61fd0a7f70878)
[2021-10-07T19:56:19-4ab3c10/illumos-amd64](https://build.golang.org/log/19591278fad57dc2558f72dda612d8148355ea4a)
[2021-10-07T18:58:33-7286502/illumos-amd64](https://build.golang.org/log/508265890fcf5a5dce31d85a3a292f97bf1667b6)
[2021-10-07T18:45:53-77f2750/illumos-amd64](https://build.golang.org/log/24a96d3c2257174fbb6e95695602647eebd58158)
[2021-10-07T18:34:24-2043b3b/illumos-amd64](https://build.golang.org/log/2d8df8475cc0371dd08fdd71b873f40f72b71f82)
[2021-10-07T18:09:53-4607ebc/illumos-amd64](https://build.golang.org/log/c409893a63b84017b85a4c61b6dc5771e5b8400f)
[2021-10-07T18:02:14-019ad98/illumos-amd64](https://build.golang.org/log/6bba569bd3997947bf09bf7df58f3ac9ebbe8db6)

**Comment by bcmills** [comment]
@jclulow, it looks like `/opt/go/bootstrap/bin/go` on this builder is an earlier build of `go1.18`. That's not going to work: the meaning of the current `go1.X` build tag changes as the release is developed, so an earlier development build of `go1.18`  will try (and fail) to build files tagged `//go:build go1.18` that use APIs added later during development.

The bootstrap Go compiler really needs to be a stable release of some sort. (Maybe install Go 1.17.2 instead?)

```
$ mote run /opt/go/bootstrap/bin/go version
go version devel go1.18-a2a9a7b513 Tue Aug 17 21:34:32 2021 +0000 illumos/amd64
```

**Comment by dmitshur** [comment]
Thanks very much @jclulow!

The issue affecting `cmd/cgo` and causing a visible failure was fixed via [CL 354695](https://golang.org/cl/354695), which is why this issue is closed now, but updating the `illumos-amd64` builder to use a stable Go 1.16 or 1.17 release will still be helpful and appreciated, as it can avoid similar problems in the future and improve our confidence in the test results. Thanks again!

**Comment by jclulow** [comment]
Thanks for continuing to build/test on illumos!  We're a small community but we very much appreciate it.

OK, I believe I have sorted that out:

```
root@gobuild1:~# /opt/go/bootstrap/bin/go version
go version go1.17.2 illumos/amd64
```

Please let me know if there are any other issues with the builder and I'll get right on it.

**Comment by jclulow** [comment]
Oh no!  I will take a look.

**Comment by jclulow** [comment]
OK, so, the first 1.17.2 bits were built on my Linux workstation using the bootstrap script, with a clone of the tree at the 1.17.2 tag.  I'm not sure why those don't work.

I've replaced them with a native (on illumos) build of 1.17.2 from the official source archive at golang.org.  When I built it, it told me `ALL TESTS PASSED` so hopefully it is alright!

I'll check in on the build jobs tomorrow.  Sorry about the mess!
